src/Campaigns/Routes/UpdateCampaign.php

```diff
@@ -60,18 +61,34 @@ public function registerRoute()
      */
     public function handleRequest(WP_REST_Request $request): WP_REST_Response
     {
+        $statusMap = [
+            'draft' => CampaignStatus::DRAFT(),
+            'active' => CampaignStatus::ACTIVE(),
+        ];
+
         foreach ($request->get_params() as $key => $value) {
-            if ('id' !== $key) {
-                $this->campaign->setAttribute($key, $value);
+            switch ($key) {
+                case 'id':
+                    break;
+                case 'status':
+                    $status = array_key_exists($value, $statusMap)
+                        ? $statusMap[$value]
+                        : CampaignStatus::DRAFT();
+
+                    $this->campaign->setAttribute('status', $status);
```

Comment:
The `validate_callback` must be a pure function. Having side effects like setting a property on the controller is not safe. This may already be broken by things like the Batch API, and could break further in the future.

This isn't the exact line that this happens on, but I can't commend on the `validate_callback` declaration.

Reply:
👆  @kjohnson @glaubersilva

Reply:
You shouldn't need the callback at all if the type is set to `integer`. But yeah you'd fetch the model in the controller callback.

Reply:
Resolved: https://github.com/impress-org/givewp/pull/7547/commits/e9253fbcda04ea19489be095af0493448c6d8c2c